.travis.yml

```diff
@@ -1,30 +1,31 @@
+dist: xenial
 sudo: required
 
-language: bash
-
-env:
-    - FLAKE8=false
-    - FLAKE8=true
+language: python
+python:
+  - "2.7"
 
 install:
     # Dependencies
     - sudo apt-get -qq update
     - sudo pip install --upgrade -qq pip
-    - sudo apt-get -qq install cdparanoia cdrdao flac gir1.2-glib-2.0 libcdio-dev libiso9660-dev libsndfile1-dev python-cddb python-gi python-musicbrainzngs python-mutagen python-setuptools sox swig libcdio-utils
-    - sudo pip install pycdio==0.21 requests
+    - sudo apt-get -qq install cdparanoia cdrdao flac libgirepository1.0-dev gir1.2-glib-2.0 libiso9660-dev libsndfile1-dev sox swig libcdio-utils
 
-    # Testing dependencies
-    - sudo apt-get -qq install python-twisted-core
-    - sudo pip install flake8
+    # Travis' Ubuntu is very old, libcdio needs to be compiled
+    - wget https://ftp.gnu.org/gnu/libcdio/libcdio-2.0.0.tar.gz
+    - tar -xvzf libcdio-2.0.0.tar.gz
+    - pushd libcdio-2.0.0
+    - ./configure
+    - sudo make install
+    - popd
 
     # Build bundled C utils
-    - cd src
+    - pushd src
     - sudo make install
-    - cd ..
+    - popd
+      #
+    # Testing dependencies
+    - sudo pip install tox-travis
 
-    # Installing
-    - sudo python setup.py install
 
-script:
-    - if [ ! "$FLAKE8" = true ]; then python -m unittest discover; fi
-    - if [ "$FLAKE8" = true ]; then flake8 --benchmark --statistics; fi
+script: tox
```

Comment:
This approach fails to take advantage of Travis' [parallel building](https://docs.travis-ci.com/user/speeding-up-the-build/) and that the whole of the test will either fail or pass, rather than (currently) code style check and unit tests pass or fail independently of each other.

Reply:
I agree. I don't think that linting takes so much time that the shaving of time is substantial, it saves resources too as the installation of dependencies only happens once.

In any case, once testing for several python interpreters is enabled in this file these will happen on parallel.

I can look on how to run linting and unittesting in parallel